services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -686,9 +686,13 @@ public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_P
         }
 
         ClientModel client = realm.getClientByClientId(clientId);
-        if (client == null || !client.isEnabled()) {
+        if (client == null) {
+            throw new WebApplicationException(
+                ErrorResponse.error(clientId + " doesn't exist", Status.BAD_REQUEST));
```

Comment:
Is it possible to change it to generic error message like "Client doesn't exists" and eventually add a warning/error message to the log with clientId? Just thinking about some potential attacks when clientId contains for example XSS script and the error will be then somehow displayed in the browser - you remember from security training :) AFAIK we changed most of error messages already to be "Generic" and not return the input, but not yet this one.

Reply:
Good one! I'm on it.